instrumentation/events/lttng-module/net.h

```diff
@@ -8,7 +8,106 @@
 #include <linux/skbuff.h>
 #include <linux/netdevice.h>
 #include <linux/ip.h>
+#include <linux/ipv6.h>
+#include <linux/tcp.h>
 #include <linux/version.h>
+#include <lttng-endian.h>
+#include <net/sock.h>
+
+#ifndef ONCE_LTTNG_NET_H
+#define ONCE_LTTNG_NET_H
+
+static inline unsigned char __has_network_hdr(struct sk_buff *skb) {
+	return skb_network_header(skb) != skb->head;
+}
+
+static struct lttng_event_field emptyfields[] = {
+};
+
+static struct lttng_event_field tcpfields[] = {
+	[0] = {
+		.name = "source",
+		.type = __type_integer(uint16_t, 0, 0, 0, __BIG_ENDIAN, 10, none),
+	},
+	[1] = {
+		.name = "dest",
+		.type = __type_integer(uint16_t, 0, 0, 0, __BIG_ENDIAN, 10, none),
+	},
+	[2] = {
+		.name = "seq",
+		.type = __type_integer(uint32_t, 0, 0, 0, __BIG_ENDIAN, 16, none),
+	},
+	[3] = {
+		.name = "ack_seq",
+		.type = __type_integer(uint32_t, 0, 0, 0, __BIG_ENDIAN, 16, none),
+	},
+	[4] = {
+		.name = "flags",
+		.type = __type_integer(uint16_t, 0, 0, 0, __BIG_ENDIAN, 16, none),
+	},
+	[5] = {
+		.name = "window",
+		.type = __type_integer(uint16_t, 0, 0, 0, __BIG_ENDIAN, 10, none),
+	},
+	[6] = {
+		.name = "check",
+		.type = __type_integer(uint16_t, 0, 0, 0, __BIG_ENDIAN, 16, none),
+	},
+	[7] = {
+		.name = "urg_ptr",
+		.type = __type_integer(uint16_t, 0, 0, 0, __BIG_ENDIAN, 16, none),
+	},
+};
+
+static struct lttng_event_field transport_fields[] = {
+	[0] = {
+		.name = "none",
+		.type = {
+			.atype = atype_struct,
+			.u._struct.nr_fields = 0,
+			.u._struct.fields = emptyfields,
+		}
+	},
+	[1] = {
+		.name = "tcp",
+		.type = {
+			.atype = atype_struct,
+			.u._struct.nr_fields = ARRAY_SIZE(tcpfields),
+			.u._struct.fields = tcpfields,
+		}
+	},
+};
+
+enum transport_header_types {
+	TH_NONE,
+	TH_TCP,
+};
+
+static inline unsigned char __get_transport_header_type(struct sk_buff *skb)
+{
+	/* If the header is not set yet, the network header will point to the head */
+	if (__has_network_hdr(skb)) {
+		/* when both transport and network header are set, transport header
```

Comment:
/*
 * When both transport and network header are set, transport header
 * is greater than network header, otherwise it points to head.
 */